components/Wallet/Replace.tsx

```diff
@@ -0,0 +1,245 @@
+import { useRouter } from 'next/router'
+import { Message } from '@glif/filecoin-message'
+import { BigNumber, FilecoinNumber } from '@glif/filecoin-number'
+import { useWallet, useWalletProvider } from '@glif/wallet-provider-react'
+import {
+  getMaxGasFee,
+  useGetMessage,
+  useGetReplaceMessageGasParams,
+  InputV2,
+  Dialog,
+  ErrorBox,
+  ShadowBox,
+  StandardBox,
+  Transaction,
+  LoginOption
+} from '@glif/react-components'
+
+import { navigate } from '../../utils/urlParams'
+import { PAGE } from '../../constants'
+import { logger } from '../../logger'
+
+export const Replace = ({ strategy }: ReplaceProps) => {
+  const router = useRouter()
+  const wallet = useWallet()
+  const cid = router.query.cid as string
+  const { loginOption, walletProvider, walletError, resetWalletError } =
+    useWalletProvider()
+
+  // Input states
+  const [expert, setExpert] = useState<boolean>(false)
+  const [gasPremium, setGasPremium] = useState<FilecoinNumber | null>(null)
+  const [gasLimit, setGasLimit] = useState<FilecoinNumber | null>(null)
+  const [gasFeeCap, setGasFeeCap] = useState<FilecoinNumber | null>(null)
+  const [isGasPremiumValid, setIsGasPremiumValid] = useState<boolean>(false)
+  const [isGasLimitValid, setIsGasLimitValid] = useState<boolean>(false)
+  const [isGasFeeCapValid, setIsGasFeeCapValid] = useState<boolean>(false)
+  const inputsValid = isGasPremiumValid && isGasLimitValid && isGasFeeCapValid
+
+  // Sending states
+  const [isSending, setIsSending] = useState<boolean>(false)
+  const [sendError, setSendError] = useState<Error | null>(null)
+
+  // Load data
+  const {
+    message,
+    loading: messageLoading,
+    error: messageError
+  } = useGetMessage(cid)
+  const {
+    gasParams,
+    loading: gasParamsLoading,
+    error: gasParamsError
+  } = useGetReplaceMessageGasParams(walletProvider, message, false)
+  const {
+    gasParams: minGasParams,
+    loading: minGasParamsLoading,
+    error: minGasParamsError
+  } = useGetReplaceMessageGasParams(walletProvider, message, true)
+
+  // Data states
+  const hasLoadError = !!(messageError || gasParamsError || minGasParamsError)
+  const isLoading = messageLoading || gasParamsLoading || minGasParamsLoading
+  const isLoaded = !!(message && gasParams && minGasParams)
+
+  // Calculate maximum transaction fee
+  const maxFee = useMemo<FilecoinNumber | null>(() => {
+    return gasLimit && gasFeeCap ? getMaxGasFee(gasFeeCap, gasLimit) : null
+  }, [gasLimit, gasFeeCap])
+
+  // Set default gas params after loading
+  // the data or disabling expert mode
+  useEffect(() => {
+    if (!expert && gasParams) {
+      setGasPremium(gasParams.gasPremium)
+      setGasLimit(gasParams.gasLimit)
+      setGasFeeCap(gasParams.gasFeeCap)
+    }
+  }, [expert, gasParams])
+
+  // Attempt sending message
+  const onSend = async () => {
+    setIsSending(true)
+    setSendError(null)
+    resetWalletError()
+    const replace = strategy === ReplaceStrategy.CANCEL
```

Comment:
Might consider renaming to `cancel` for clarity

Reply:
Whoops, yeah that should definitely be called `cancel`!